### PR TITLE
ENH: Add DeformAVolumeWithAThinPlateSpline

### PR DESCRIPTION
**Pull request by thewtex**
Adapted from:

  https://itk.org/Doxygen/html/Examples_2RegistrationITKv4_2ThinPlateSplineWarp_8cxx-example.html

However, we use itk mesh IO to read the point sets, and use the reference image interface to ResampleImageFilter. Both were added since that example was created.

### CODE REVIEW

**Review by tbirdso — APPROVED** [review]
Looks great!

File: src/Core/Transform/DeformAVolumeWithAThinPlateSpline/Documentation.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
+:name: DeformAVolumeWithAThinPlateSpline
+
+Deform a Volume With a Thin Plate Spline
+================================
```
Comment (jhlegarreta) [inline src/Core/Transform/DeformAVolumeWithAThinPlateSpline/Documentation.rst:4]: Warning reported by the CI build:
https://open.cdash.org/viewBuildError.php?type=1&buildid=8380872

**Review by jhlegarreta — COMMENTED** [review]
Thanks for doing this Matt.

An in-line comment about the build failure.

File: src/Core/Transform/DeformAVolumeWithAThinPlateSpline/Documentation.rst (hunk 1 above)
Comment (thewtex) [reply src/Core/Transform/DeformAVolumeWithAThinPlateSpline/Documentation.rst:4]: :eagle: :eye:

Comment (jhlegarreta) [reply src/Core/Transform/DeformAVolumeWithAThinPlateSpline/Documentation.rst]: Wonderful. Thanks Matt :green_heart:. Resolving.

**Review by jhlegarreta — APPROVED** [review]

File: src/Core/Transform/DeformAVolumeWithAThinPlateSpline/Code.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+# limitations under the License.
+
+import argparse
+import itk
+import numpy as np
+
+parser = argparse.ArgumentParser(
+    description="Deform a Volume With a Thin Plate Spline."
+)
+parser.add_argument("source_landmarks")
```
Comment (orena1) [inline src/Core/Transform/DeformAVolumeWithAThinPlateSpline/Code.py:24]: Is there a way to direct to the specification of the landmark file?

**Review by orena1 — COMMENTED** [review]
Sorry for been late, how can I know the specification of the landmarks files?

**Pull request merged**